### PR TITLE
Create "SuperSet" rule

### PR DESCRIPTION
**Pull request by Singwai**

### CODE REVIEW

**Comment by Singwai** [comment]
👀  will fix the issue in a moment

File: library/Rules/SuperSet.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class SuperSet extends AbstractRule
+{
+
+    public $compareTo;
+
+    /**
+     * SuperSet constructor.
+     * @param $compareTo
+     */
+    public function __construct($compareTo)
```
Comment (henriquemoody) [inline library/Rules/SuperSet.php]: `array $compareTo`

File: library/Rules/SuperSet.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        $this->compareTo = $compareTo;
+    }
+
+    public function validate($input)
+    {
+        return !array_diff($input, $this->compareTo);
```
Comment (henriquemoody) [inline library/Rules/SuperSet.php]:
```php
if (!is_array($input)) {
    return false;
}
```

File: tests/unit/Rules/SuperSetTest.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+namespace Respect\Validation\Rules;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group  rule
+ * @covers \Respect\Validation\Rules\SuperSet
+ * @covers \Respect\Validation\Exceptions\SuperSetException
+ */
+class SuperSetTest extends TestCase
```
Comment (henriquemoody) [inline tests/unit/Rules/SuperSetTest.php]: Could you extend [RuleTestCase](https://github.com/Respect/Validation/blob/1.1/tests/library/Rules/RuleTestCase.php) instead? Check this out: https://github.com/Respect/Validation/blob/1.1/CONTRIBUTING.md#creating-unit-tests

File: library/Rules/SuperSet.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Singwai Chan<c.singwai@gmail.com>
```
Comment (henriquemoody) [inline library/Rules/SuperSet.php]: This header should be the same for all files.

File: library/Rules/SuperSet.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class SuperSet extends AbstractRule
```
Comment (henriquemoody) [inline library/Rules/SuperSet.php]: You can add here a:
```php
/**
 * @author Singwai Chan <c.singwai@gmail.com>
 */
```

**Comment by henriquemoody** [comment]
I sent #1045 with the changes that I request. Also while refactoring that, I noticed that the name of the rule didn't indicate what it was validating. So I renamed it to **Subset** instead.

Thanks for contributing! 🐼

**Pull request closed without merging**